components/brave_ads/core/internal/catalog/catalog_json_reader.cc

```diff
@@ -93,6 +94,14 @@ absl::optional<CatalogInfo> ReadCatalog(const std::string& json) {
       const bool success = base::StringToDouble(value, &creative_set.value);
       DCHECK(success);
 
+      if (creative_set_node.HasMember("embedding")) {
+        std::vector<float> embedding;
```

Comment:
can we not just add to creative_set.embedding rather than creating a new vector and copying that vector?